File: ObjectSearchWorld/src/searchWorld/src/Pomcp/Location.java
package searchWorld.src.Pomcp;

import searchWorld.src.State.SearchState;

/**
 * Created by awandzel on 7/17/18.
 */
public class Location implements Comparable<Location>{
  public Integer x;
  public Integer y;

  public Location(Integer nx, Integer ny) {
    this.x = nx;
    this.y = ny;
  }

  public Location(Location l) {
    this.x = l.x;
    this.y = l.y;
  }

  public Location copy(){
    return new Location(this.x, this.y);
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) return true; // If they have the same address in memory
    if (o == null || getClass() != o.getClass()) return false;
    if (!(o instanceof Location)) return false;

    Location that = (Location) o;

    return (that.x == this.x) && (that.y == this.y);
  }

  @Override
  public int hashCode() {
    return 37 * this.x + this.y;
  }

  @Override
  public String toString() {
    return "x=" + this.x + ", " + "y=" + this.y;
  }

  @Override
  public int compareTo(Location other) {
    return Integer.compare((37 * this.x)+this.y, (37*other.x)+other.y);
  }
}

File: Multi_Object_Search/Pomdp/Location.py
'''
    Created by awandzel on 10/30/18.
'''

class Location:
    def __init__(self, x, y):
        self.x = x
        self.y = y

    def __hash__(self):
        return 37 * (self.x + self.y)

    def __eq__(self, other):
        if other == None  or self.__class__ != other.__class__: return False
        if not isinstance(other, Location): return False

        return other.x == self.x and other.y == self.y







    def __str__(self):
        return "x=" + str(self.x) + ", " + "y=" + str(self.y)





